### ISSUE TITLE
v0.2.1 compilation fails with `duplicate symbol _got_a_configuration` on Mac

### ISSUE THREAD

**Issue by EricFromCanada** [mywave82/opencubicplayer#15]
v0.2.1 [fails to compile on macOS](https://github.com/Homebrew/homebrew-core/pull/49858): 
```
clang -g -O2 -fno-common -fPIC -Wall -Itimidity-git/interface -Itimidity-git/libarc -Itimidity-git/libunimod -Itimidity-git/timidity -Itimidity-git/utils -I. -DHAVE_CONFIG_H -I.././ -dynamiclib -flat_namespace -undefined suppress -o playtimidity.dylib timiditypchan.o timidityplay.o timiditypplay.o timiditypdots.o ocp-output.o timidity-git/interface/wrdt_dumb.o timidity-git/libarc/arc.o timidity-git/libarc/arc_mime.o timidity-git/libarc/arc_lzh.o timidity-git/libarc/arc_tar.o timidity-git/libarc/arc_zip.o timidity-git/libarc/deflate.o timidity-git/libarc/explode.o timidity-git/libarc/inflate.o timidity-git/libarc/unlzh.o timidity-git/libarc/url.o timidity-git/libarc/url_b64decode.o timidity-git/libarc/url_cache.o timidity-git/libarc/url_dir.o timidity-git/libarc/url_file.o timidity-git/libarc/url_hqxdecode.o timidity-git/libarc/url_inflate.o timidity-git/libarc/url_mem.o timidity-git/libarc/url_pipe.o timidity-git/libarc/url_qsdecode.o timidity-git/libarc/url_uudecode.o timidity-git/timidity/aq.o timidity-git/timidity/audio_cnv.o timidity-git/timidity/common.o timidity-git/timidity/effect.o timidity-git/timidity/filter.o timidity-git/timidity/freq.o timidity-git/timidity/instrum.o timidity-git/timidity/loadtab.o timidity-git/timidity/mfi.o timidity-git/timidity/miditrace.o timidity-git/timidity/mix.o timidity-git/timidity/mt19937ar.o timidity-git/timidity/optcode.o timidity-git/timidity/quantity.o timidity-git/timidity/rcp.o timidity-git/timidity/readmidi.o timidity-git/timidity/recache.o timidity-git/timidity/resample.o timidity-git/timidity/reverb.o timidity-git/timidity/sbkconv.o timidity-git/timidity/sffile.o timidity-git/timidity/sfitem.o timidity-git/timidity/smfconv.o timidity-git/timidity/smplfile.o timidity-git/timidity/sndfont.o timidity-git/timidity/tables.o timidity-git/timidity/timidity.o timidity-git/timidity/version.o timidity-git/timidity/wrd_read.o timidity-git/timidity/wrdt.o timidity-git/utils/mblock.o timidity-git/utils/fft4g.o timidity-git/utils/memb.o timidity-git/utils/nkflib.o timidity-git/utils/strtab.o timidity-git/utils/timer.o -lm
duplicate symbol _got_a_configuration in:
    timidityplay.o
    timidity-git/timidity/timidity.o
ld: 1 duplicate symbol for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[1]: *** [playtimidity.dylib] Error 1
make: *** [playgmi/gmiptype.o] Error 2
make: *** Waiting for unfinished jobs....
duplicate symbol _got_a_configuration in:
    timidityplay.o
    timidity-git/timidity/timidity.o
ld: 1 duplicate symbol for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[1]: *** [playtimidity.dylib] Error 1
make: *** [dirs] Error 2
```

**Comment by mywave82** [comment]
Before I dive into this. Does the stock TiMidity+ compile ok?

**Comment by mywave82** [comment]
master has been patched. Can you please try. Might be other similar issues with other symbols

**Comment by EricFromCanada** [comment]
With the above change:
```
clang -g -O2 -fno-common -fPIC -Wall -I.././ -dynamiclib -flat_namespace -undefined suppress -o playhvl.dylib loader.o player.o hvlpchan.o hvlpdots.o hvlpinst.o hvlpplay.o hvlplay.o hvlpt
rak.o
duplicate symbol _ht in:
    hvlpchan.o
    hvlpdots.o
duplicate symbol _plInstUsed in:
    hvlpchan.o
    hvlpdots.o
duplicate symbol _ht in:
    hvlpchan.o
    hvlpinst.o
duplicate symbol _plInstUsed in:
    hvlpchan.o
    hvlpinst.o
duplicate symbol _ht in:
    hvlpchan.o
    hvlpplay.o
duplicate symbol _plInstUsed in:
    hvlpchan.o
    hvlpplay.o
duplicate symbol _ht in:
    hvlpchan.o
    hvlplay.o
duplicate symbol _plInstUsed in:
    hvlpchan.o
    hvlplay.o
duplicate symbol _ht in:
    hvlpchan.o
    hvlptrak.o
duplicate symbol _plInstUsed in:
    hvlpchan.o
    hvlptrak.o
ld: 10 duplicate symbols for architecture x86_64
```

**Comment by mywave82** [comment]
Can you please retest again the latest master

**Comment by EricFromCanada** [comment]
Looks like that fixed it. Did you have a date in mind for the next release?

**Comment by mywave82** [comment]
I'm head deep into sidplayfp, after that I plan a release. Going to need compile test of that. Some few weeks from now I hope

**Comment by mywave82** [comment]
@EricFromCanada can you please compiletest the latest master, and test to play a .SID file? Example file to test http://hvsc.csdb.dk/MUSICIANS/H/Huelsbeck_Chris/Great_Giana_Sisters.sid
(This plugin can currently potentially contain linking problems)

**Comment by EricFromCanada** [comment]
```
...
/Applications/Xcode.app/Contents/Developer/usr/bin/make -C playsid TOPDIR=.././
clang -g -O2 -fno-common -fPIC -Wall -I.././ sidptype.c -o sidptype.o -c
clang -g -O2 -fno-common -fPIC -Wall -I.././ -dynamiclib -flat_namespace -undefined suppress  -o sidtype.dylib sidptype.o
clang++ -g -O2 -fno-common -fPIC -Wall -I.././ cpiinfo.cpp -o cpiinfo.o -c 
clang++ -std=c++11 -DHAVE_CONFIG_H=1 -DSIDTUNE_NO_STDIN_LOADER=1 -DPACKAGE_NAME=\"sidplayfp\" -DVERSION=\"2.0.1\" -DPACKAGE_VERSION=\"2.0.1-1\" -DPACKAGE_URL=\"\" -I.. -g -O2 -fno-common -fPIC -I./sidplayfp-config -I./sidplayfp-git/libsidplayfp/src sidplay.cpp -o sidplay.o -c
clang++ -g -O2 -fno-common -fPIC -Wall -I.././ sidpplay.cpp -o sidpplay.o -c
sidpplay.cpp:56:10: fatal error: 'sidplayfp/SidTuneInfo.h' file not found
#include <sidplayfp/SidTuneInfo.h>
         ^
1 error generated.
make[1]: *** [sidpplay.o] Error 1
make: *** [dirs] Error 2

HOMEBREW_VERSION: 2.2.6-54-ge88eea6
ORIGIN: https://github.com/Homebrew/brew.git
HEAD: e88eea64129b68cdec24e0d82522b8a4ad2d8646
Last commit: 2 hours ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: ac6d14452798aa937ef88f9441516eb31d378597
Core tap last commit: 85 minutes ago
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CACHE: /Users/eric/Library/Caches/Homebrew
HOMEBREW_LOGS: /Users/eric/Library/Logs/Homebrew
CPU: octa-core 64-bit haswell
Homebrew Ruby: 2.6.3 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/bin/ruby
Clang: 8.0 build 800
Git: 2.25.0 => /usr/local/opt/git/bin/git
Curl: 7.43.0 => /usr/bin/curl
Java: 1.8.0_111, 1.6.0_65-b14-468
macOS: 10.11.6-x86_64
CLT: 8.2.0.0.1.1480973914
Xcode: 8.2.1
XQuartz: 2.7.11 => /opt/X11

HOMEBREW_CC: clang
HOMEBREW_CXX: clang++
CMAKE_PREFIX_PATH: /usr/local
CMAKE_INCLUDE_PATH: /usr/include/libxml2:/System/Library/Frameworks/OpenGL.framework/Versions/Current/Headers
CMAKE_LIBRARY_PATH: /System/Library/Frameworks/OpenGL.framework/Versions/Current/Libraries
PKG_CONFIG_PATH: /usr/local/opt/libogg/lib/pkgconfig:/usr/local/opt/flac/lib/pkgconfig:/usr/local/opt/libvorbis/lib/pkgconfig:/usr/local/opt/mad/lib/pkgconfig
PKG_CONFIG_LIBDIR: /usr/lib/pkgconfig:/usr/local/Homebrew/Library/Homebrew/os/mac/pkgconfig/10.11
HOMEBREW_GIT: /usr/local/opt/git/bin/git
ACLOCAL_PATH: /usr/local/share/aclocal
PATH: /usr/local/Homebrew/Library/Homebrew/shims/mac/super:/usr/local/opt/xa/bin:/usr/local/opt/flac/bin:/usr/bin:/bin:/usr/sbin:/sbin
```

**Comment by mywave82** [comment]
Did you update git submodules and rerun configure?

./configure will tell you which git command to run if a submodule is missing

**Comment by mywave82** [comment]
test now. I had some left-over files of libsidplayfp in my /usr/local/include

**Comment by EricFromCanada** [comment]
Progress:
```
...
clang++ -std=c++11 -DHAVE_CONFIG_H=1 -DSIDTUNE_NO_STDIN_LOADER=1 -DPACKAGE_NAME=\"sidplayfp\" -DVERSION=\"2.0.1\" -DPACKAGE_VERSION=\"2.0.1-1\" -DPACKAGE_URL=\"\" -g -O2 -fno-common -fPIC -I./sidplayfp-config -I./sidplayfp-git/libsidplayfp/src -o libsidplayfp-sidplayfp-SidTuneInfo.o sidplayfp-git/libsidplayfp/src/sidplayfp/SidTuneInfo.cpp -c
clang++ -std=c++11 -DHAVE_CONFIG_H=1 -DSIDTUNE_NO_STDIN_LOADER=1 -DPACKAGE_NAME=\"sidplayfp\" -DVERSION=\"2.0.1\" -DPACKAGE_VERSION=\"2.0.1-1\" -DPACKAGE_URL=\"\" -g -O2 -fno-common -fPIC -I./sidplayfp-config -I./sidplayfp-git/libsidplayfp/src -o libsidplayfp-sidtune-p00.o sidplayfp-git/libsidplayfp/src/sidtune/p00.cpp -c
clang++ -std=c++11 -DHAVE_CONFIG_H=1 -DSIDTUNE_NO_STDIN_LOADER=1 -DPACKAGE_NAME=\"sidplayfp\" -DVERSION=\"2.0.1\" -DPACKAGE_VERSION=\"2.0.1-1\" -DPACKAGE_URL=\"\" -g -O2 -fno-common -fPIC -I./sidplayfp-config -I./sidplayfp-git/libsidplayfp/src -o libsidplayfp-sidtune-prg.o sidplayfp-git/libsidplayfp/src/sidtune/prg.cpp -c
o65file=`echo sidplayfp-git/libsidplayfp/src/sidtune/sidplayer1.bin | sed 's/bin/o65/'`; \
	xa -R -G sidplayfp-git/libsidplayfp/src/sidtune/sidplayer1.a65 -o 65file && \
	od -v -An -w8 -tx1 65file | sed -re 's/[[:alnum:]]+/0x&,/g' > sidplayfp-git/libsidplayfp/src/sidtune/sidplayer1.bin && \
	rm -f 65file
od: illegal option -- w
usage: od [-aBbcDdeFfHhIiLlOosvXx] [-A base] [-j skip] [-N length] [-t type]
          [[+]offset[.][Bb]] [file ...]
/usr/bin/sed: illegal option -- r
usage: sed script [-Ealn] [-i extension] [file ...]
       sed [-Ealn] [-i extension] [-e script] ... [-f script_file] ... [file ...]
make[1]: *** [sidplayfp-git/libsidplayfp/src/sidtune/sidplayer1.bin] Error 1
make: *** [dirs] Error 2
```

**Comment by mywave82** [comment]
Notes for myself:

man-page for MAC version of od:
https://www.oreilly.com/library/view/macintosh-terminal-pocket/9781449328962/re24.html

sed:
 -r adds support for extended syntax
need to test which part of the expression that does not work, and try to fix it

**Comment by mywave82** [comment]
Updated the Makefile to use hexdump instead now in git master

**Comment by mywave82** [comment]
so, it all compiles well? :-)

**Comment by EricFromCanada** [comment]
It compiled and ran, but only output noise when attempting to play the mentioned .SID file.

**Comment by mywave82** [comment]
Trying to get a VirtualBox up and running with OS-X. Go an image booting, but now requires an AppleID to continue ;-)

**Comment by EricFromCanada** [comment]
You can skip that.

**Comment by mywave82** [comment]
Was able to get brew installed, and the dependencies, and build Open Cubic Player... but

VirtualBox with macOS as guest, on non macOS host, does not seem to work for audio...

**Comment by EricFromCanada** [comment]
Supposedly [MultiBeast](http://incomingtech.weebly.com/how-to-enable-sound-on-mac-os-x-in-virualbox.html) can rectify this.

**Comment by mywave82** [comment]
Done several attempts with MultiBeast, and most of the versions fails to "Build".

Until somebody can provide a working virtual machine image, or provide remote access (TeamViewer for instance), I can not fix problems specific for this platform

**Comment by pyrate01** [comment]
Hey, same bug / different platform (Manjaro 20.0.3 Kernel 5.6.19-2) 

When I attempt to install ocp from the AUR (ocp 0.2.1-1)....

`/usr/bin/ld: timidity-git/timidity/timidity.o:(.bss+0x140): multiple definition of `got_a_configuration'; timidityplay.o:(.bss+0x8c4): first defined here
collect2: error: ld returned 1 exit status
make[1]: *** [Makefile:185: playtimidity.so] Error 1
make[1]: Leaving directory '/var/tmp/pamac-build-pyrate/ocp/src/opencubicplayer/playgmi'
make: *** [Makefile:155: dirs] Error 2
`
Have been enjoying ocp for years, I appreciate your dedication to this ongoing project. Will be around for any needed assistance.

**Comment by mywave82** [comment]
@pyrate01: This should be fixed in 0.2.2 already. I keep this bug open until I have access to OSX and check/fix a couple of other issues (The CoreAudio native driver)
